ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -657,14 +657,24 @@ static KsqlRestApplication buildApplication(
 
     final String commandTopicName = ReservedInternalTopics.commandTopic(ksqlConfig);
 
+    CommandTopicBackup commandTopicBackup = new CommandTopicBackupNoOp();
+    if (!ksqlConfig.getString(KsqlConfig.KSQL_METASTORE_BACKUP_LOCATION).isEmpty()) {
+      commandTopicBackup = new CommandTopicBackupImpl(
+          ksqlConfig.getString(KsqlConfig.KSQL_METASTORE_BACKUP_LOCATION),
+          commandTopicName
+      );
+    }
+    final CommandTopicBackup finalCommandTopicBackup = commandTopicBackup;
```

Comment:
Is this `finalCommandTopicBackup` necessary? Seems you can use the same `commandTopicBackup` variable. It will have either the NoOp or the Impl instances.

Reply:
I think that was leftover from when I was passing the backup to the CommandRunner, which was later changed so that CommandRunner checks corruption from the CommandStore instead. 

I'll move this entire chunk of creating the backup back into the CommandStore again